packages/smooth_app/lib/pages/product/product_page/footer/new_product_footer_settings.dart

```diff
@@ -49,8 +49,12 @@ class ProductFooterSettingsButton extends StatelessWidget {
           title: appLocalizations.product_page_action_bar_setting_modal_title,
           prefixIndicator: true,
           closeButton: true,
-          expandBody: true,
-          body: const _ProductActionBarModal(),
+          expandBody: false, // Avoid unbounded expansion
```

Comment:
If I were you I wouldn't recode this part with your personal tastes.
The point here is not to change the way it used to be, the point is to put it how it was before.
The only thing you should do here is say "btw it's dynamic content" (or "not static", whatever), with an additional parameter.